chain/chain/src/flat_storage_resharder.rs

```diff
@@ -181,6 +181,22 @@ impl FlatStorageResharder {
         info!(target: "resharding", ?split_params, "initiating flat storage shard split");
 
         self.check_new_event_is_allowed()?;
+
+        // Get all resharding blocks, if any, from the existing scheduled split shard,
+        // and merge them with the new resharding block.
+        let mut resharding_blocks = vec![resharding_block];
+        if let Some(FlatStorageReshardingEventStatus::SplitShard(parent, params, _)) =
+            self.resharding_event()
+        {
+            if parent_shard == parent
+                && params.left_child_shard == left_child_shard
+                && params.right_child_shard == right_child_shard
+                && params.shard_layout == *shard_layout
```

Comment:
Why do you need those checks?

Reply:
These are not strictly necessary. The checks are a safeguard to discard the old candidate resharding block if the shard layout of the new resharding is different. It can't really happen today since shard layout is fixed.